edk2toolext/environment/extdeptypes/nuget_dependency.py

```diff
@@ -245,6 +248,12 @@ def _attempt_nuget_install(self, install_dir, non_interactive=True):
             if non_interactive and found_cred_provider:  # we should be interactive next time
                 self._attempt_nuget_install(install_dir, False)
             else:
+                if is_unauthorized:
+                    logging.warning("[Nuget] A package requires credentials, but you do not have a credential "\
```

Comment:
Is it possible they do have a credential provider installed but the credentials are incorrect?

Trying to make sure the message is not helpful but potentially misleading.

Reply:
I've fixed the logic for when to print these warnings. These warnings are only for the situation where the user is not using a credential provider but is receiving a 401 error (unauthorized).

The scenario you mention above does not result in my message being printed. However out of curiosity, I did test what would happen if I logged into the wrong account for the credential provider. It just failed with the normal warning (`WARNING - [SDE] Failed to fetch NugetDependecy: ... `). On the second pass it pops up an error message allowing you to sign out, and sign back in with a different account.

Reply:
Thanks for checking, looks good with the latest commit.